### DISCUSSION TITLE
[QUESTION]  "dmenu_path like" on rofi?

### DISCUSSION THREAD

**Discussion by thealio** [davatorium/rofi#1778] [Q&A]
Hi,
as written on the title
I'm writing a polkit script (I'll post it later, currently I haven't my PC with me) and I'm using  dmenu_path on it.
I'd like to use all rofi instead of dmenu.
So, there is an equivalent command to dmenu_path  on rofi?

**Comment by DaveDavenport** [comment]
What is dmenu_path?

**Reply by thealio** [reply]
https://manpages.ubuntu.com/manpages/bionic/man1/dmenu_path.1.html

**Comment by thealio** [comment]
https://manpages.ubuntu.com/manpages/bionic/man1/dmenu_path.1.html

**Reply by DaveDavenport** [reply]
rofi does not have that. It scans $PATH on each startup. This been shown to be quick enough.
Why would you want another implementation for this to begin with? It seems pretty agnostic to the used interface.

**Reply by thealio** [reply]
I'm adapting a dmenu script to rofi

**Reply by thealio** [reply]
I'll post the script later
Maybe in the end another solution can be adopted using rofi

**Reply by DaveDavenport** [reply]
Its a tiny script, https://git.suckless.org/dmenu/file/dmenu_path.html  just do something similar in your script.

Rofi does not use a cache for $PATH and there is no tool to see them.

**Reply by thealio** [reply]
Interesting
I'll check later if this can be adapted to the script, and I'll report here
Thank you

**Reply by thealio** [reply]
Hi
here 's my work
[rofi-polkit.zip](https://github.com/davatorium/rofi/files/10406530/rofi-polkit.zip)

**Reply by thealio** [reply]
now I have to
- merge all the 3 scripts into a single one
- adapt the script for Thunar custom actions
- modify the script for make it to work as a GUI prompt

Of course, help will be appreciated
Also, opinions, suggestions, hints etc. will be appreciated too

I'm getting in troubles with these 3 steps

**Comment by thealio** [comment]
Hi @DaveDavenport 
I write the thunar custom action

SUDO_ASKPASS=~/.scripts/rofi/rofi-polkit/rofi-sudopass sudo -A  xdg-open %f

It works pretty good

I hope that my code could be useful
